src/tests/end2end/AddTests.cpp

```diff
@@ -44,9 +44,10 @@ TEST_F(AddTests, AddConstantAndInput) {
         -0.03482966, -0.7343786,  -0.76851964, 0.9446942,  -0.35489243, 0.44452578,  0.00648887,
         -0.55656946, -0.735903,   0.22050636,  -0.5008282, -1.3132697,  1.6642882,   -0.48397836,
         0.20099205,  -0.28786168, 1.3315053,   -0.41619393};
-    const ml::Input input = {dataA.data(), dataA.size() * sizeof(float)};
-    const ml::Result result = utils::AwaitCompute(graph, {{"a", input}}).Get("c");
-    EXPECT_TRUE(utils::CheckShape(result, {3, 4, 5}));
+    const ml::Resource input = {(void*)dataA.data(), dataA.size() * sizeof(float)};
+    std::vector<float> result(utils::SizeOfShape({3, 4, 5}));
+    const ml::Resource output = {result.data(), result.size() * sizeof(float)};
+    utils::Compute(graph, {{"a", input}}, {{"c", output}});
```

Comment:
Could we pass the std::vector<T> to the compute method?
```c++
utils::Compute(graph, {{"a", dataA}}, {{"c", result}});
```

Reply:
It can be simplified if we don't care about the usage of  the `Compute` API align with Spec, but It isn't good to guide developer how to use WebNN API.

[webnn-polyfill ](https://github.com/webmachinelearning/webnn-polyfill/blob/master/test/ops/add.js#L44)also doesn't simply the usage.
```
const inputs = {a: new Float32Array([])};
const outputs = {c: new Float32Array(utils.sizeOfShape([3, 4, 5]))};
graph.compute(inputs, outputs);
```

Reply:
Yes, it's better to simplify.

Reply:
There is a tradeoff. It is important to guide developers to use `compute` API in spec. However it also helps that we share the pattern of the common usage of the API. We already have `utils::Compute` that is for this purpose. Here my point is we provide an utility that helps set the `Resource` (or `ArrayBufferView`) as my other comment) from a `std::vector`. Developers may find this useful and even use the utility in their code, e.g., @MarkGHX reuse the utils::Compute in his OpenCV.js WebNN project.

Reply:
Got it.